src/nvim/ex_cmds.c

```diff
@@ -224,15 +225,17 @@ void ex_align(exarg_T *eap)
          */
         if (has_tab)
           while (new_indent > 0) {
-            (void)set_indent(new_indent, 0);
+            assert(new_indent <= INT_MAX);
+            (void)set_indent((int)new_indent, 0);
             if (linelen(NULL) <= width) {
               /*
                * Now try to move the line as much as possible to
                * the right.  Stop when it moves too far.
                */
-              do
-                (void)set_indent(++new_indent, 0);
-              while (linelen(NULL) <= width);
+              do {
+                assert(new_indent + 1 <= INT_MAX);
+                (void)set_indent((int)++new_indent, 0);
```

Comment:
`++var` in any position where its value is used is valid because in any position where its value is used it *is* meaningful. According to style guide postincrement should be used in *statements* and this is not the one (statement here is `(void)set_indent((int)++new_indent, 0);` and not just `++new_indent`).